contrib/test.sh

```diff
@@ -11,6 +11,15 @@ if cargo --version | grep nightly; then
     NIGHTLY=true
 fi
 
+# On MacOS on 1.41 we get a link failure with syn
+# This is fixed by https://github.com/rust-lang/rust/pull/91604 (I think)
+# but implies that we can't do testing on MacOS for now, at least with 1.41.
+if cargo --version | grep "1\.41\.0"; then
```

Comment:
This seems to be working but we are running with toolchain 1.41.1, why is the regex statement succeeding?

Reply:
This shows it works: `Continuous integration / Tests (macOS-latest, 1.41.1) (pull_request) Successful in 15s `

Reply:
Bizzare, this is the version of cargo used in the CI pipeline: cargo 1.41.0 (626f0f40e 2019-12-03)

Reply:
It "works" because we `exit 0` here.

Reply:
lolz, I meant why does `cargo --version` report "1\.41\.0" when we are using the 1.41.1 toolchain.

Reply:
Said another way, the regex surprised me because of the zero in the toolchain version.

Reply:
Oh, hmm, I wonder if this `grep` is doing the opposite of what I want (and skipping every test)

Reply:
Oops, I forgot to reply to this... the reason is that **cargo** is at 1.41.0 even though rustc is at 1.41.1.

Reply:
oooh, I see. Thanks